### PR TITLE
feat: client side query builder search

### PR DESCRIPTION
**Pull request by ahmadshaheer**
### Summary
build client-side search with features of Query Builder, it can accept predefined keys and values

#### Screenshots
with `attributeKeys` only:
https://www.loom.com/share/6beeb6ea1a514ec484ecba40bfda6c46

with `attributeKeys` and `attributeValuesMap`
https://www.loom.com/share/2d5b781fdefd4c4ea9f623f1a40da50b

#### Affected Areas and Manually Tested Areas

- Alert history page

### CODE REVIEW

File: frontend/src/components/ClientSideQBSearch/ClientSideQBSearchDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+interface ICustomDropdownProps {
+	menu: React.ReactElement;
+}
+
+export default function ClientSideQBSearchDropdown(
+	props: ICustomDropdownProps,
+): React.ReactElement {
+	const { menu } = props;
+	return <div className="">{menu}</div>;
```
Comment (SagarRajput-7) [inline frontend/src/components/ClientSideQBSearch/ClientSideQBSearchDropdown.tsx:9]: empty className

File: frontend/src/components/ClientSideQBSearch/ClientSideQBSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
+import {
+	IBuilderQuery,
+	TagFilter,
+} from 'types/api/queryBuilder/queryBuilderData';
+import { popupContainer } from 'utils/selectPopupContainer';
+import { v4 as uuid } from 'uuid';
+
+import ClientSideQBSearchDropdown from './ClientSideQBSearchDropdown';
+
+export interface AttributeKey {
```
Comment (SagarRajput-7) [inline frontend/src/components/ClientSideQBSearch/ClientSideQBSearch.tsx:59]: this must be present already

Comment (ahmadshaheer) [reply frontend/src/components/ClientSideQBSearch/ClientSideQBSearch.tsx:59]: @SagarRajput-7, for this, all the other fields except `key` are optional. I guess it'll be better to get rid of the optional fields as we are only getting the `key` in the API response.

File: frontend/src/components/ClientSideQBSearch/ClientSideQBSearchDropdown.tsx (hunk 1 above)
Comment (ahmadshaheer) [reply frontend/src/components/ClientSideQBSearch/ClientSideQBSearchDropdown.tsx:9]: Done, I got rid of the dropdown renderer component, as it's not necessary, IMO.

File: frontend/src/pages/AlertDetails/hooks.tsx [hunk 3]
```diff
@@ -277,6 +282,7 @@ export const useGetAlertRuleDetailsTimelineTable = (): GetAlertRuleDetailsTimeli
 			timelineFilter,
 			updatedOrder,
 			offset,
+			JSON.stringify(filters.items),
```
Comment (YounixM) [inline frontend/src/pages/AlertDetails/hooks.tsx:285]: When can this fail?

Comment (ahmadshaheer) [reply frontend/src/pages/AlertDetails/hooks.tsx:285]: @YounixM, IMO, it is not going to fail. if there are no filters, we're assigning `initialFilters` to filters.

https://github.com/SigNoz/signoz/blob/9e2386008ee52468c0ce0e49e10f3728bebdf503/frontend/src/constants/queryBuilder.ts#L159-L162

https://github.com/SigNoz/signoz/blob/9e2386008ee52468c0ce0e49e10f3728bebdf503/frontend/src/container/AlertHistory/Timeline/Table/Table.tsx#L16-L25

https://github.com/SigNoz/signoz/blob/9e2386008ee52468c0ce0e49e10f3728bebdf503/frontend/src/pages/AlertDetails/hooks.tsx#L257-L261

**Review by rahulkeswani101 — COMMENTED** [review]
@ahmadshaheer Can you check why the build is failing?
Rest LGTM.

**Review by rahulkeswani101 — DISMISSED** [review]

**Review by SagarRajput-7 — DISMISSED** [review]

**Review by rahulkeswani101 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**